### DISCUSSION TITLE
@id1402 asks:  How do you set the date time from month, day, year values?

### DISCUSSION THREAD

**Discussion by Makuna** [Makuna/Rtc#169] [Q&A] [answered]
@id1402 From your question you posted on a [pull request](https://github.com/Makuna/Rtc/pull/38#issuecomment-1526214114) (not good practice BTW, ask here and reference the pull request is the proper way to do it).

If you have values for month/day/year and you want to instance a RtcDateTime, it is done like this...
```
    RtcDateTime myTime(2023, 4, 27, 19, 33, 00);
```
if you have no plans to use `myTime` other than calling `SetDateTime()` then you can save the step of creating a local variable like this...
```
   rtc.SetDateTime(RtcDateTime(2023, 4, 27, 19, 33, 00));
```
If you have a string like you formatted ("Apr 25 2023, 00:28:09"); then you will have to parse the string into the values as there is no automatic parser for all the date/time formats that exists.   There is currently only one method, `InitWithIso8601()` which takes a string formatted like "Sat, 06 Dec 2009 12:34:56"; where the hours is 24 hours.

**Comment by id1402** [comment]
>@id1402 From your question you posted on a https://github.com/Makuna/Rtc/pull/38#issuecomment-1526214114 (not good practice BTW, ask here and reference the pull request is the proper way to do it).

Sorry.

```
RtcDateTime myTime(2023, 4, 27, 19, 33, 00);
```
What variable type? RtcDateTime myTime(uint16_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t)?
Why double zero "00" in seconds? Are other values than "00" working?
Must there be leading zeroe in other values?

Thank you for the library and your help!

**Reply by Makuna** [reply]
should have been just 0 not double 0.  It's just an example, put whatever seconds you want to set it too.  There is no parameter validation, so you could even pass 200 seconds, which makes an invalid time and the call to `IsValid()` will return false.

I don't understand your question "What variable type?".  Date and time in this library are always represented by the type `RtcDateTime`.  The example constructs one using the Constructor that takes arguments of each date and time unit.
```
RtcDateTime(uint16_t year,
        uint8_t month,
        uint8_t dayOfMonth,
        uint8_t hour,
        uint8_t minute,
        uint8_t second)
```

also note, day of the week is calculated so no need to set that value so it's not an argument to the constructor.

**Comment by id1402** [comment]
>so you could even pass 200 seconds, which makes an invalid time and the call to IsValid() will return false.

So the time cannot be set to wrong values?

>I don't understand your question "What variable type?".

I ask because i make:
```
RtcDateTime timeNow(dateYear, dateMonth, dateDay, clockHour, clockMinute, clockSecond);
rtc.SetDateTime(timeNow);
```
So it's good to know the appropriate data type for the used variables.

**Reply by Makuna** [reply, marked as answer]
To make sure the time is correct, check it like...

```
RtcDateTime timeNow(dateYear, dateMonth, dateDay, clockHour, clockMinute, clockSecond);
if (timeNow.IsValid()) {
    rtc.SetDateTime(timeNow);
}
else {
    // display error?
}
```

**Comment by id1402** [comment]
That's great! Thank you.
